syntax = "proto3";
package tinyim;

option cc_generic_services = true;

message UserAndIdNum {
    int64 user_id = 1;
    int64 need_msgid_num = 2;
}

message MsgIdRequest {
    repeated UserAndIdNum user_ids = 1;  // first one is sender
}

message MsgIds {
    int64 user_id = 1;
    int64 start_msg_id = 2;
    int64 msg_id_num = 3;
}

message MsgIdReply {
    repeated MsgIds msg_ids = 1;
}

service IdGenService {
    // 获取id，支持批量获取, 多用户，多用户的多id
    rpc IdGenerate(MsgIdRequest) returns (MsgIdReply);
}
